src/realm/replication.cpp

```diff
@@ -62,36 +64,323 @@ Replication::version_type Replication::prepare_commit(version_type orig_version)
+        const char* collection_type = "";
+        if (col_key.is_collection()) {
+            if (col_key.is_list()) {
+                collection_type = "list ";
+            }
+            else if (col_key.is_dictionary()) {
+                collection_type = "dictionary ";
+            }
+            else {
+                collection_type = "set ";
+            }
+        }
+        if (target_table) {
+            logger->log(util::Logger::Level::debug, "On class '%1': Add property '%2' %3linking '%4'",
+                        t->get_class_name(), col_name, collection_type, target_table->get_class_name());
+        }
+        else {
+            logger->log(util::Logger::Level::debug, "On class '%1': Add property '%2' %3of %4", t->get_class_name(),
+                        col_name, collection_type, type);
+        }
+    }
+    select_table(t);                  // Throws
+    m_encoder.insert_column(col_key); // Throws
+}
+
+void Replication::erase_column(const Table* t, ColKey col_key)
+{
+    if (auto logger = get_logger()) {
+        logger->log(util::Logger::Level::debug, "On class '%1': Remove property '%2'", t->get_class_name(),
+                    t->get_column_name(col_key));
+    }
+    select_table(t);                 // Throws
+    m_encoder.erase_column(col_key); // Throws
+}
+
 void Replication::create_object(const Table* t, GlobalKey id)
 {
+    if (auto logger = get_logger()) {
+        logger->log(util::Logger::Level::debug, "Create object '%1'", t->get_class_name());
+    }
     select_table(t);                              // Throws
     m_encoder.create_object(id.get_local_key(0)); // Throws
 }
 
-void Replication::create_object_with_primary_key(const Table* t, ObjKey key, Mixed)
+void Replication::create_object_with_primary_key(const Table* t, ObjKey key, Mixed pk)
 {
+    if (auto logger = get_logger()) {
+        logger->log(util::Logger::Level::debug, "Create object '%1' with primary key %2", t->get_class_name(), pk);
+    }
     select_table(t);              // Throws
     m_encoder.create_object(key); // Throws
 }
 
+void Replication::remove_object(const Table* t, ObjKey key)
+{
+    if (auto logger = get_logger()) {
+        if (t->is_embedded()) {
+            logger->log(util::Logger::Level::debug, "Remove embedded object '%1'", t->get_class_name());
+        }
+        else if (t->get_primary_key_column()) {
+            logger->log(util::Logger::Level::debug, "Remove object '%1' with primary key %2", t->get_class_name(),
+                        t->get_primary_key(key));
+        }
+        else {
+            logger->log(util::Logger::Level::debug, "Remove object '%1'", t->get_class_name());
+        }
+    }
+    select_table(t);              // Throws
+    m_encoder.remove_object(key); // Throws
+}
+
+inline void Replication::select_obj(ObjKey key)
+{
+    if (key != m_selected_obj) {
+        if (auto logger = get_logger()) {
+            if (logger->would_log(util::Logger::Level::debug)) {
```

Comment:
shouldn't this be at trace level as well?

Reply:
The idea should be that we log mutations on objects at debug level, but get the values as well at trace level.